dev-docs/RFCs/v5.0/attribute-animation-rfc.md

```diff
@@ -9,41 +9,128 @@ Notes:
-
-Allowing an accessor to be set to a value instead of a function could be a nice way to enable generic vertex attributes. I.e.
+In deck.gl layer classes, certain attributes can declare themselves to be "animatable", which allows for smooth interpolation when their values are updated. This will be desirable for attributes such as sizes, widths, positions and colors. Animation will be an opt-in feature for backward compatibility:
 ```js
-   new Layer({getColor: x => [255, 0, 0]}) // would create a full array/WebGLBuffer with colors, each set to [255, 0, 0], just like today
+this.state.attributeManager.add({
+  // Not animatable
+  pickingColors: {size: 3, type: GL.UNSIGNED_BYTE, update: this.calculatePickingColors},
+  // Animatable
+  positions: {size: 3, accessor: 'getPosition', update: this.calculatePositions, animate: true},
+  colors: {size: 4, type: GL.UNSIGNED_BYTE, accessor: 'getColor', update: this.calculateColors, animate: true}
+});
 ```
-while
+
+When creating layers, animation can be enabled by supplying an `transition` prop. Animation parameters are defined per attribute by using attribute names or accessor names as keys, similar to that of `updateTriggers`:
 ```js
-   new Layer({getColor: [255, 0, 0]}) // would set a generic vertex attribute (one value shared by all verts, not allocating any array/buffers at all)
+new Layer({
+  transition: {
+    getPositions: {
+      duration: 600
+    },
+    getColors: {
+      duration: 300,
+      easing: d3.easeCubicInOut
+    }
+  }
+});
 ```
 
-Likewise:
+| Parameter | Type     | Description |
+| --------- | -------- | ----------- |
+| duration  | Number   | Duration of the transition animation, in milliseconds |
+| easing    | Function | Easing function that maps a value from [0, 1] to [0, 1], see http://easings.net/ |
+| onComplete | Function   | Callback when the transition is done |
+
+
+### Implementation
+
+- Layers will get the feature (almost) free:
+    + The animation states will be tracked in the attribute manager.
+    + The base `Layer` class will ensure that the animated attributes are updated every render cycle before the `draw` call.
+- In WebGL2 enabled browsers, use TransformFeedback for interpolation:
+    + The inputs are from and to value arrays, and current ratio (float between 0, 1) as a uniform
+    + The output is passed to the vertex shader for the target attribute
+- In non-WebGL2 enabled browsers, we either drop the feature, or fall back to interpolation on the CPU.
+
+### TransformFeedback vs VertexShader
+
+VertexShader pros:
+- Available in all supported browsers
+
+VertexShader cons:
+- Must rewrite existing layers’ vertex shaders to double each animatable attribute and perform interpolation before other calculations. The cost of a heavier shader and double attribute upload applies even if transition is disabled on a layer.
```

Comment:
>  double attribute upload applies even if transition is disabled 

Not necessarily. One can set an attribute to a single value (constant) that applies to all vertices (which you mention in the ogther file in this PR) We could use that technique for attributes that are not used.